chirho/dynamical/handlers/interruption.py

```diff
@@ -130,5 +130,17 @@ def __init__(
         self.observation = observation
         super().__init__(times)
 
-    def _pyro_post_simulate(self, msg: dict) -> None:
-        self.trajectory = observe(self.trajectory, self.observation)
+    def _pyro_simulate(self, msg: dict) -> None:
```

Comment:
I don't think you should need to change `StaticBatchObservation` at all, other than adding a single `super()._pyro_post_simulate` call:
```
def _pyro_post_simulate(self, msg: dict) -> None:
    super()._pyro_post_simulate(msg)  # update self.trajectory
    self.trajectory = observe(self.trajectory, self.observation)